### DISCUSSION TITLE
ConnectorSkin add label not refresh in view

### DISCUSSION THREAD

**Discussion by syluna** [eckig/graph-editor#197] [Q&A] [answered]
In my GConnectorSkin in the constructor I create Polygon with the direction and I want to add a label with dynamic calculation position - - from top the label is under and centered
- from rigth the label is on the left
- from left the label is on the right
- from botton the label is up
To do this in the previous version (1.4) I set label and dynamic calculation of insets: 

> double SIZE = 11;
> root.resize(SIZE, SIZE);
> // Initialize and calculate polygone for the render
> // ...
> // Now I draw Label out of element but relative to connector
> SimpleObjectProperty<Insets> inset = new SimpleObjectProperty<>(new Insets(0));
> 				label.setText(cnx.getDisplay());
> 				label.setManaged(true);
> 				label.getStyleClass().add(cnx.getStyle());
> 				label.setStyle("-fx-font-size: 9; -fx-alignment: center-left;  -fx-text-alignment: left;");
> 				label.setTextOverrun(OverrunStyle.CLIP);
> 				label.paddingProperty().bind(inset);
> 				label.widthProperty().addListener((o, ov, nv) -> {
> 					if (new Insets(0).equals(label.getPadding())) {
> 						inset.set(switch (connector.getType()) {
> 						case DefaultConnectorTypes.TOP_INPUT, DefaultConnectorTypes.TOP_OUTPUT ->
> 							new Insets(SIZE, (-SIZE - nv.doubleValue()) / 2, 0, (SIZE - nv.doubleValue()) / 2);
> 						case DefaultConnectorTypes.RIGHT_INPUT, DefaultConnectorTypes.RIGHT_OUTPUT ->
> 							new Insets(SIZE / 2 - 8, 3, 0, -nv.doubleValue() - 3);
> 						case DefaultConnectorTypes.BOTTOM_INPUT, DefaultConnectorTypes.BOTTOM_OUTPUT -> new Insets(
> 								-SIZE / 2 - 10, (-SIZE - nv.doubleValue()) / 2, 0, (SIZE - nv.doubleValue()) / 2);
> 						case DefaultConnectorTypes.LEFT_INPUT, DefaultConnectorTypes.LEFT_OUTPUT ->
> 							new Insets(SIZE / 2 - 8, 0, 0, SIZE + 3);
> 						default -> new Insets(1);
> 						});
> 					}
> 				});
> 				root.getChildren().add(label);

At first the connector is draw with partial label 
![image](https://github.com/eckig/graph-editor/assets/1336540/e1f2acd6-c64e-45e7-9d12-be7b57d7ebc2)

When I simply clic in Node the full label is redraw (the width event is already call at the initialization, not recall when I click. And the skin is correctly draw : 
![image](https://github.com/eckig/graph-editor/assets/1336540/eafe3bbb-56a2-4048-adb3-1794c8de53d3)

Have you any idea what I miss ?  
I'm on this probleme for tow day and now idea why... 
Thanks

**Comment by eckig** [comment, marked as answer]
I can not really reproduce this, as this is not a standalone example, but I would try to simplify things: Maybe you could just use a normal `Label` and put the text and triangle next to each other. The position of the triangle relative to the text can then be set with the `contentDisplayProperty()`.

**Reply by syluna** [reply]
Thanks eckig, I try this. I hadn't thought of treating it like that.

**Reply by syluna** [reply]
This work like a charm.
